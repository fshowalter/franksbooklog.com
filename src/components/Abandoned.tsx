import type React from "react";

export function Abandoned({
  className,
  value,
}: {
  className?: string;
  value: string | undefined;
}): false | React.JSX.Element {
  if (value !== "Abandoned") {
    return false;
  }

  return (
    <div
      className={`
        rounded-sm bg-abandoned px-2 py-1 font-sans text-xxs font-bold
        tracking-prose text-inverse uppercase
        ${className}
      `}
    >
      Abandoned
    </div>
  );
}
