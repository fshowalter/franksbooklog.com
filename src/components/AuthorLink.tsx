import type { ElementType } from "react";

type Props = {
  as?: ElementType;
  className?: string;
  name: string;
  notes?: string;
  slug?: string;
};

export function AuthorLink({
  as = "div",
  className,
  name,
  notes,
  slug,
}: Props): JSX.Element {
  const Component = as;

  if (slug) {
    return (
      <Component className={className}>
        <a href={`/reviews/authors/${slug}/`}>{name}</a>
        <AuthorNotes notes={notes} />
      </Component>
    );
  }

  return (
    <Component className={className}>
      {name}
      <AuthorNotes notes={notes} />
    </Component>
  );
}

function AuthorNotes({ notes }: { notes?: string }) {
  if (!notes) {
    return false;
  }

  return <span className="text-subtle"> ({notes})</span>;
}
