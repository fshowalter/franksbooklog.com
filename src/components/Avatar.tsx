import type { AvatarImageProps } from "~/api/avatars";

type AvatarProps = React.ImgHTMLAttributes<HTMLImageElement> & {
  className?: string;
  fill?: string;
  height: number;
  imageProps: AvatarImageProps | undefined;
  loading: "eager" | "lazy";
  width: number;
};

export function Avatar({
  className,
  fill,
  imageProps,
  ...rest
}: AvatarProps): JSX.Element {
  if (imageProps) {
    return <img {...imageProps} alt="" {...rest} className={className} />;
  }

  return (
    <div className={className}>
      <svg
        fill={fill || "var(--bg-stripe)"}
        viewBox="0 0 16 16"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M16 8A8 8 0 110 8a8 8 0 0116 0zm-6-3a2 2 0 11-4 0 2 2 0 014 0zM8 9a5 5 0 00-4.546 2.916A5.986 5.986 0 008 14a5.986 5.986 0 004.546-2.084A5 5 0 008 9z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  );
}
