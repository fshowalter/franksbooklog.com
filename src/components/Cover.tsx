import React from "react";

import type { CoverImageProps } from "~/api/covers";

type Props = {
  decoding: "async" | "auto" | "sync";
  height: number;
  imageProps: CoverImageProps | undefined;
  loading: "eager" | "lazy";
  width: number;
} & React.ImgHTMLAttributes<HTMLImageElement>;

export function Cover({
  decoding = "async",
  imageProps,
  loading = "lazy",
  ...rest
}: Props): JSX.Element {
  return (
    <img
      {...imageProps}
      {...rest}
      alt=""
      decoding={decoding}
      loading={loading}
      style={{ aspectRatio: "0.66666667" }}
    />
  );
}
