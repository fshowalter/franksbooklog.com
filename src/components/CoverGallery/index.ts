export { Cover } from "./Cover";
export { CoverGallery } from "./CoverGallery";
