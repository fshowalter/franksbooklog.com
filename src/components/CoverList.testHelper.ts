import type { UserEvent } from "@testing-library/user-event";

import { screen } from "@testing-library/react";

export async function clickShowMore(user: UserEvent) {
  await user.click(screen.getByText("Show More"));
}

export function getGroupedCoverList() {
  return screen.getByTestId("grouped-cover-list");
}
