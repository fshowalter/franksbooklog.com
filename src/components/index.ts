export * from "./Article";
export * from "./Author";
export * from "./Authors";
export * from "./HeadBuilder";
export * from "./Home";
export * from "./Readings";
export * from "./Review";
export * from "./Reviews";
export * from "./Shelf";
export * from "./Stats";
