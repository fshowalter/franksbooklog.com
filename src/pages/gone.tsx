import { graphql } from "gatsby";
import { Article, HeadBuilder } from "../components";

export function Head(): JSX.Element {
  return (
    <HeadBuilder
      pageTitle="410: Gone"
      description="Forget it, Jake. It's Chinatown."
      image={null}
      article
    />
  );
}

export default function GonePage({
  data,
}: {
  data: Queries.GonePageQuery;
}): JSX.Element {
  return (
    <Article
      image={data.backdrop}
      alt="Jake Gittes walks away."
      article={data.page}
    />
  );
}

export const pageQuery = graphql`
  query GonePage {
    backdrop: file(absolutePath: { regex: "/backdrops/gone.png$/" }) {
      ...ArticleBackdrop
    }
    page: markdownRemark(frontmatter: { slug: { eq: "gone" } }) {
      ...ArticleData
    }
  }
`;
