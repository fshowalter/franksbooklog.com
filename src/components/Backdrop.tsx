import type { AvatarImageProps } from "~/api/avatars";
import type { BackdropImageProps } from "~/api/backdrops";

import { Avatar } from "./Avatar";

export function AvatarBackdrop({
  avatarImageProps,
  backdropImageProps,
  breadcrumb,
  deck,
  name,
}: {
  avatarImageProps: AvatarImageProps | undefined;
  backdropImageProps: BackdropImageProps;
  breadcrumb?: React.ReactNode;
  deck: React.ReactNode;
  name: string;
}) {
  const heroImage = (
    <img
      className={`
        absolute inset-0 size-full object-cover object-top text-inverse blur-xs
      `}
      {...backdropImageProps}
      {...BackdropImageConfig}
      alt=""
      fetchPriority="high"
      loading="eager"
    />
  );

  return (
    <Wrapper
      centerText={true}
      heroImage={heroImage}
      size="auto"
      textInverse={true}
    >
      {avatarImageProps && (
        <div
          className={`
            mx-auto mt-4 mb-6 w-4/5 max-w-[250px] transform-gpu overflow-hidden
            ${avatarImageProps && "rounded-full shadow-all"}
          `}
        >
          <Avatar
            data-pagefind-meta="image[src]"
            decoding="async"
            height={250}
            imageProps={avatarImageProps}
            loading="lazy"
            width={250}
          />
        </div>
      )}
      <Breadcrumb value={breadcrumb} />
      <Title center={true} value={name} />
      <Deck center={true} value={deck} />
    </Wrapper>
  );
}

export function BreadcrumbLink({
  children,
  color = "inverse",
  href,
}: {
  children: React.ReactNode;
  color?: "default" | "inverse";
  href: string;
}) {
  return (
    <a
      className={`
        font-sans text-sm tracking-wide uppercase underline-offset-8
        ${
          color === "inverse"
            ? `
              text-inverse decoration-inverse-subtle decoration-2
              hover:underline
            `
            : `
              text-subtle
              hover:text-accent
            `
        }
      `}
      href={href}
    >
      {children}
    </a>
  );
}

function Breadcrumb({ value }: { value?: React.ReactNode }) {
  if (!value) {
    return false;
  }

  return (
    <p className="mb-2 font-sans text-sm tracking-wide text-subtle uppercase">
      {value}
    </p>
  );
}

function Deck({
  center,
  shadow,
  subtle,
  value,
}: {
  center?: boolean;
  shadow?: boolean;
  subtle?: boolean;
  value?: React.ReactNode;
}) {
  if (!value) {
    return false;
  }

  return (
    <p
      className={`
        mt-1 text-base
        desktop:my-4 desktop:text-xl
        ${center ? `text-center` : ""}
        ${shadow ? `[text-shadow:1px_1px_2px_black]` : ""}
        ${subtle ? `text-subtle` : ""}
      `}
    >
      {value}
    </p>
  );
}

function Title({
  center,
  className,
  value,
}: {
  center?: boolean;
  className?: string;
  value: string;
}) {
  return (
    <h1
      className={
        className ||
        `
          font-sans
          ${center ? "text-center" : ""}
          text-3xl font-bold tracking-widest uppercase
          tablet:text-4xl
          tablet-landscape:text-5xl
          desktop:text-7xl
        `
      }
    >
      {value}
    </h1>
  );
}

function Wrapper({
  centerText = false,
  children,
  heroImage,
  narrowChildren = false,
  size = "default",
  textInverse = false,
}: {
  centerText?: boolean;
  children: React.ReactNode;
  heroImage?: React.ReactNode;
  narrowChildren?: boolean;
  size?: "auto" | "default" | "large" | "small" | "xsmall";
  textInverse?: boolean;
}) {
  const defaultSizes =
    "min-h-[340px] tablet:min-h-[640px] desktop:min-h-[clamp(640px,70vh,1350px)]";

  const largeSizes = "min-h-[90vh] max-h-[1350px]";

  const smallSizes = "min-h-[clamp(340px,50vh,1350px)]";

  const xsmallSizes = "min-h-[clamp(340px,40vh,1350px)]";

  const sizes =
    size === "large"
      ? largeSizes
      : size === "small"
        ? smallSizes
        : size === "xsmall"
          ? xsmallSizes
          : size === "auto"
            ? "min-h-[25vw]"
            : defaultSizes;

  return (
    <header
      className={`
        ${sizes}
        ${textInverse ? "text-inverse" : ""}
        ${heroImage ? "bg-[#000]" : "bg-canvas"}
        relative flex w-full flex-col content-start items-center justify-end
        gap-6 pt-20
        laptop:pt-24
      `}
    >
      {heroImage}
      <div
        className={`
          ${centerText ? "items-center" : ""}
          z-10 mx-auto flex w-full
          ${
            narrowChildren
              ? `
                px-container text-center
                tablet:max-w-none tablet:px-0
              `
              : `max-w-(--breakpoint-desktop) px-container`
          }
          flex-col
          ${
            heroImage
              ? `
                after:absolute after:top-0 after:left-0 after:-z-10 after:h-full
                after:w-full after:bg-linear-to-t after:from-[rgba(0,0,0,.85)]
                after:to-50%
                tablet:after:to-25%
              `
              : ""
          }
          py-8
          tablet:pb-10
          laptop:pt-10
          desktop:pt-20
        `}
      >
        {children}
      </div>
    </header>
  );
}

export const BackdropImageConfig = {
  height: 1350,
  sizes: "100vw",
  width: 2400,
};

export function Backdrop({
  blur = false,
  breadcrumb,
  centerText = false,
  deck,
  imageProps,
  size = "default",
  title,
  titleClasses,
}: {
  blur?: boolean;
  breadcrumb?: React.ReactNode;
  centerText?: boolean;
  deck?: React.ReactNode;
  imageProps: BackdropImageProps;
  size?: "default" | "large" | "small";
  title: string;
  titleClasses?: string;
}) {
  const heroImage = (
    <img
      className={`
        absolute inset-0 size-full object-cover object-top
        ${blur && `blur-xs`}
      `}
      {...imageProps}
      {...BackdropImageConfig}
      alt=""
      fetchPriority="high"
      loading="eager"
    />
  );

  return (
    <Wrapper
      heroImage={heroImage}
      narrowChildren={centerText}
      size={size}
      textInverse={true}
    >
      <Breadcrumb value={breadcrumb} />
      <Title className={titleClasses} value={title} />
      <Deck shadow={true} value={deck} />
    </Wrapper>
  );
}
