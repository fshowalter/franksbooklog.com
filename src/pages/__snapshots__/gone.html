<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" sizes="any" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <link rel="manifest" href="/site.webmanifest" />
    <link rel="sitemap" href="/sitemap-index.xml" />
    <meta name="description" content="Forget it, Jake. It's Chinatown." />
    <meta
      name="og:image"
      content="/_image?/content%2Fassets%2Fbackdrops%2Fgone.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&#38;w=1200&#38;h=675&#38;q=80&#38;f=jpeg"
    />
    <meta property="og:url" content="https://www.franksbooklog.com//" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="410: Gone" />
    <meta
      property="og:description"
      content="Forget it, Jake. It's Chinatown."
    />
    <title>410: Gone | Frank&#39;s Book Log</title>
  </head>
  <body>
    <div>
      <a
        href="#content"
        class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
        >Skip to content</a
      >
      <div class="flex min-h-full w-full flex-col bg-default">
        <header
          class="z-20 flex w-full items-center justify-between px-container py-4 tablet:p-6 desktop:inset-x-0 desktop:z-40 desktop:flex-row desktop:flex-wrap desktop:px-16 desktop:py-8 desktop:text-left"
          style="position: absolute"
        >
          <div class="flex flex-col">
            <div
              class="whitespace-nowrap font-normal leading-8"
              style="font-size: 1.5625rem"
            >
              <a href="/">Frank&#x27;s Book Log</a>
            </div>
            <p class="w-full pl-px text-sm italic leading-4 opacity-85">
              Literature is a relative term.
            </p>
          </div>
          <div class="flex items-center">
            <nav class="hidden w-full desktop:block desktop:w-auto">
              <ul class="flex flex-wrap justify-start gap-x-6 text-xl">
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/"
                    >Home</a
                  >
                </li>
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/how-i-grade/"
                    >How I Grade</a
                  >
                </li>
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/reviews/"
                    >Reviews</a
                  >
                </li>
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/readings/"
                    >Reading Log</a
                  >
                </li>
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/authors/"
                    >Authors</a
                  >
                </li>
                <li class="tracking-serif-wide block whitespace-nowrap">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/shelf/"
                    >Shelf</a
                  >
                </li>
              </ul>
            </nav>
            <div class="z-[1000]">
              <button
                data-open-modal="true"
                disabled=""
                aria-label="Search"
                aria-keyshortcuts="Control+K"
                class="flex items-center justify-between overflow-hidden text-sm leading-6 ring-default desktop:ml-6"
                type="button"
                title="Search: Control+K"
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke-width="1.5"
                  stroke="currentColor"
                  class="size-6 desktop:size-7"
                >
                  <path
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
                  ></path>
                </svg>
              </button>
            </div>
            <input type="checkbox" id="mobile-nav" class="hidden" /><label
              for="mobile-nav"
              class="hamburger-icon desktop:hidden"
              ><span style="background: #fff" class="hamburger-icon-bars"></span
            ></label>
            <ul
              class="hamburger-menu flex flex-col items-start gap-y-5 text-left text-inverse desktop:hidden"
            >
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/how-i-grade/">How I Grade</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/reviews/">Reviews</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/readings/">Reading Log</a>
                <ol class="mt-4">
                  <li
                    class="text-inverse-subtle mb-2 font-sans text-xs uppercase tracking-wide"
                  >
                    <a href="/readings/stats/">Stats</a>
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/authors/">Authors</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/shelf/">Shelf</a>
              </li>
            </ul>
          </div>
        </header>
        <main class="grow" id="content">
          <article>
            <header
              class="min-h-[240px] desktop:min-h-[clamp(640px,50vh,1350px)] relative flex w-full flex-col content-start items-center justify-end gap-6 bg-canvas pb-8 pt-40 tablet:pb-10 tablet:pt-40 desktop:pb-16 desktop:pt-40"
            >
              <div
                class="z-10 mx-auto flex w-full max-w-screen-max flex-col px-container"
              >
                <h1
                  class="font-sans text-2xl font-bold uppercase tracking-widest desktop:text-7xl"
                >
                  410 Gone
                </h1>
                <p class="mt-1 text-base desktop:my-4 desktop:text-xl">
                  Forget it, Jake. It&#x27;s Chinatown.
                </p>
              </div>
            </header>
            <section class="flex flex-col items-center pb-32 pt-16">
              <div class="px-container">
                <div
                  class="rendered-markdown font-light text-md/7 tracking-0.3px tablet:text-xl/8 max-w-prose"
                >
                  <p>Whatever you’re looking for, it’s not coming back.</p>
                </div>
              </div>
            </section>
          </article>
          <div
            data-pagefind-ignore="true"
            class="flex w-full flex-col items-center gap-y-12 bg-subtle pb-32 pt-16 tablet:pt-8 desktop:gap-y-24"
          >
            <nav
              data-pagefind-ignore="true"
              class="relative flex w-full flex-col items-center tablet:w-auto"
            >
              <div
                class="relative mx-auto w-full max-w-screen-max px-container desktop:px-20"
              >
                <h2
                  class="py-10 font-sans text-xs font-semibold uppercase tracking-wide text-subtle"
                >
                  Recent<!-- -->
                  <a href="/reviews/" class="text-accent">Reviews</a>
                </h2>
                <ul
                  class="flex flex-wrap justify-center gap-x-[3%] gap-y-[6vw] desktop:justify-between"
                >
                  <li
                    class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                  >
                    <img
                      srcset="
                        /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif   62w,
                        /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w,
                        /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w,
                        /_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w
                      "
                      src="/_image?/content%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                      alt=""
                      width="248"
                      height="372"
                      loading="lazy"
                      decoding="async"
                      style="aspect-ratio: 0.66666667"
                    />
                    <div
                      class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                    >
                      <div class="text-center text-md font-medium leading-6">
                        <a
                          href="/reviews/recessional-by-david-mamet/"
                          rel="canonical"
                          class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                          >Recessional</a
                        >
                      </div>
                      <div
                        class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                      >
                        2022<!-- -->
                        |<!-- -->
                        <span class="whitespace-nowrap">Nonfiction</span>
                      </div>
                      <p class="text-center text-base text-subtle">
                        by
                        <!-- -->David Mamet
                      </p>
                      <picture
                        ><source
                          srcset="/svg/2-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/2-stars.svg"
                          alt="D: 2 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mt-2"
                      /></picture>
                    </div>
                  </li>
                  <li
                    class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                  >
                    <img
                      srcset="
                        /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif   62w,
                        /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w,
                        /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w,
                        /_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w
                      "
                      src="/_image?/content%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                      alt=""
                      width="248"
                      height="372"
                      loading="lazy"
                      decoding="async"
                      style="aspect-ratio: 0.66666667"
                    />
                    <div
                      class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                    >
                      <div class="text-center text-md font-medium leading-6">
                        <a
                          href="/reviews/fast-forward-by-stephen-morris/"
                          rel="canonical"
                          class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                          >Fast Forward</a
                        >
                      </div>
                      <div
                        class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                      >
                        2020<!-- -->
                        |<!-- -->
                        <span class="whitespace-nowrap">Nonfiction</span>
                      </div>
                      <p class="text-center text-base text-subtle">
                        by
                        <!-- -->Stephen Morris
                      </p>
                      <picture
                        ><source
                          srcset="/svg/4-half-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/4-half-stars.svg"
                          alt="A-: 4.5 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mt-2"
                      /></picture>
                    </div>
                  </li>
                  <li
                    class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                  >
                    <img
                      srcset="
                        /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif   62w,
                        /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w,
                        /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w,
                        /_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w
                      "
                      src="/_image?/content%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                      alt=""
                      width="248"
                      height="372"
                      loading="lazy"
                      decoding="async"
                      style="aspect-ratio: 0.66666667"
                    />
                    <div
                      class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                    >
                      <div class="text-center text-md font-medium leading-6">
                        <a
                          href="/reviews/gothic-by-philip-fracassi/"
                          rel="canonical"
                          class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                          >Gothic</a
                        >
                      </div>
                      <div
                        class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                      >
                        2023<!-- -->
                        |<!-- -->
                        <span class="whitespace-nowrap">Novel</span>
                      </div>
                      <p class="text-center text-base text-subtle">
                        by
                        <!-- -->Philip Fracassi
                      </p>
                      <picture
                        ><source
                          srcset="/svg/4-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/4-stars.svg"
                          alt="B+: 4 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mt-2"
                      /></picture>
                    </div>
                  </li>
                  <li
                    class="relative flex w-[48.5%] max-w-[248px] flex-col items-center border-default bg-default tablet:w-[22.75%]"
                  >
                    <img
                      srcset="
                        /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;q=80&amp;f=avif   62w,
                        /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;q=80&amp;f=avif 124w,
                        /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;q=80&amp;f=avif 248w,
                        /_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;q=80&amp;f=avif 496w
                      "
                      src="/_image?/content%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                      alt=""
                      width="248"
                      height="372"
                      loading="lazy"
                      decoding="async"
                      style="aspect-ratio: 0.66666667"
                    />
                    <div
                      class="flex grow flex-col items-center px-[8%] pb-8 pt-3 desktop:pl-[8.5%] desktop:pr-[10%]"
                    >
                      <div class="text-center text-md font-medium leading-6">
                        <a
                          href="/reviews/record-play-pause-by-stephen-morris/"
                          rel="canonical"
                          class="z-10 inline-block decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 hover:text-accent hover:underline hover:before:opacity-0"
                          >Record Play Pause</a
                        >
                      </div>
                      <div
                        class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                      >
                        2019<!-- -->
                        |<!-- -->
                        <span class="whitespace-nowrap">Nonfiction</span>
                      </div>
                      <p class="text-center text-base text-subtle">
                        by
                        <!-- -->Stephen Morris
                      </p>
                      <picture
                        ><source
                          srcset="/svg/5-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/5-stars.svg"
                          alt="A: 5 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mt-2"
                      /></picture>
                    </div>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
        </main>
        <footer
          class="bg-footer flex flex-wrap items-start justify-between gap-[10%] px-container py-20 text-inverse tablet:px-12 tablet:pt-10 desktop:p-20"
        >
          <div class="flex flex-col pb-12">
            <div class="flex flex-col">
              <div
                class="whitespace-nowrap font-normal leading-8"
                style="font-size: 1.5625rem"
              >
                <a href="/">Frank&#x27;s Book Log</a>
              </div>
              <p class="w-full pl-px text-sm italic leading-4 opacity-85">
                Literature is a relative term.
              </p>
            </div>
          </div>
          <div
            class="flex flex-col gap-20 pb-20 pt-10 tablet:basis-1/2 tablet:pt-20"
          >
            <ul class="flex w-full flex-col gap-y-10 text-inverse max:w-auto">
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/how-i-grade/"
                  >How I Grade</a
                >
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/reviews/">Reviews</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/readings/">Reading Log</a>
                <ol class="mt-4">
                  <li
                    class="text-inverse-subtle mb-2 font-sans text-xs uppercase tracking-wide"
                  >
                    <a class="hover:text-inverse" href="/readings/stats/"
                      >Stats</a
                    >
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/authors/">Authors</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/shelf/">Shelf</a>
              </li>
            </ul>
            <a
              href="#top"
              class="mb-8 w-full max-w-button bg-canvas py-5 text-center font-sans text-xs uppercase tracking-wide text-default hover:bg-inverse hover:text-inverse"
              >To the top</a
            >
            <p class="text-inverse-subtle w-full font-normal leading-4">
              All covers used in accordance with the<!-- -->
              <a
                href="http://www.copyright.gov/title17/92chap1.html#107"
                class="text-inherit underline decoration-2 underline-offset-4 hover:bg-default hover:text-default"
                >Fair Use Law.</a
              >
            </p>
          </div>
        </footer>
      </div>
    </div>
    <dialog
      class="relative m-0 h-screen max-h-none w-screen max-w-unset border border-default bg-subtle shadow-lg open:flex open:flex-col tablet:mx-auto tablet:mt-16 tablet:h-min tablet:max-h-[min(100vh_-_128px,900px)] tablet:min-h-60 tablet:max-w-[min(100vw_-_64px,36rem)] tablet:rounded-xl desktop:min-h-0"
      aria-label="Search"
    >
      <div data-dialog-frame class="contents">
        <button
          data-close-modal
          class="px-4 font-sans text-sm font-light hover:bg-inverse hover:text-inverse focus:outline-accent"
        >
          Cancel
        </button>
        <div class="contents"><div id="pagefind__search"></div></div>
        <div
          class="hidden px-[var(--search-dialog-padding)] py-4 font-sans text-xs font-light tracking-wide text-subtle desktop:block"
        >
          <kbd class="rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >Tab</kbd
          >
          to navigate
          <kbd
            class="ml-4 rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >ESC</kbd
          >
          to close
        </div>
      </div>
    </dialog>
    <script>
      (() => {
        const openBtn = document.querySelector("button[data-open-modal]");
        if (!openBtn) return;
        if (/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)) {
          openBtn.setAttribute("aria-keyshortcuts", "Meta+K");
          openBtn.setAttribute("title", `Search: ⌘K`);
        }
      })();
    </script>
  </body>
</html>
